lib/src/rules/use_getters_to_access_properties.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.use_getters_to_access_properties;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/standard_resolution_map.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc =
+    r'Use getters for operations that conceptually access properties.';
+
+const _details = r'''
+
+**DO** use getters for operations that conceptually access properties.
+
+**BAD:**
+```
+rectangle.getWidth()
+collection.isEmpty()
+button.canShow()
+```
+
+**GOOD:**
+```
+rectangle.width
+collection.isEmpty
+button.canShow
+```
+
+''';
+
+bool _hasSideEffect(AstNode node) {
+  return (node is MethodInvocation ||
+      node is FunctionExpressionInvocation ||
+      node is AssignmentExpression ||
+      node is PrefixExpression ||
+      node is PostfixExpression);
+}
+
+class UseGettersToAccessProperties extends LintRule {
+  _Visitor _visitor;
+  UseGettersToAccessProperties()
+      : super(
+            name: 'use_getters_to_access_properties',
+            description: _desc,
+            details: _details,
+            group: Group.style) {
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  _Visitor(this.rule);
+
+  @override
+  visitMethodDeclaration(MethodDeclaration node) {
+    if (!node.isStatic &&
+        !node.isGetter &&
+        !resolutionMap.elementDeclaredByMethodDeclaration(node).isOverride &&
```

Comment:
This only check if has @override annotation, I do not like to repeat all the code of annotate_override, but I thinks would give a much better experience for the user, how can we solve this @bwilkerson? One option is copy paste all that code, or other is to merge both rules in one file, or maybe create an abstract Visitor.